python_modules/dagster/dagster/_core/instance/runs/run_domain.py

```diff
@@ -720,11 +728,29 @@ def get_keys_to_reexecute(
             {key for key in to_reexecute if isinstance(key, AssetCheckKey)},
         )
 
+    def _get_repo_scoped_asset_node(
+        self,
+        asset_graph: "BaseAssetGraph",
+        asset_key: AssetKey,
+        remote_job_origin: Optional["RemoteJobOrigin"] = None,
+    ) -> Optional["BaseAssetNode"]:
+        from dagster._core.definitions.assets.graph.remote_asset_graph import RemoteAssetNode
+
+        if not asset_graph.has(asset_key):
+            return None
+        asset_node = asset_graph.get(asset_key)
+        if isinstance(asset_node, RemoteAssetNode):
+            return asset_node.resolve_to_repo_scoped_node(
+                check.not_none(remote_job_origin).repository_origin
+            )
+        return asset_node
+
     def log_asset_planned_events(
         self,
         dagster_run: DagsterRun,
         execution_plan_snapshot: "ExecutionPlanSnapshot",
         asset_graph: "BaseAssetGraph",
+        remote_job_origin: Optional["RemoteJobOrigin"],
```

Comment:
[1] is this not in sync with the field on `DagsterRun`?

Reply:
good point